File: app/src/main/java/com/azizbekian/spyur/injection/AppComponent.java
package com.azizbekian.spyur.injection;

import android.content.Context;

import com.azizbekian.spyur.api.ApiInteractor;
import com.bumptech.glide.RequestManager;

import javax.inject.Singleton;

import com.azizbekian.spyur.activity.ListingActivity;

import dagger.Component;

/**
 * Created on May 09, 2016.
 *
 * @author Andranik Azizbekian (andranik.azizbekyan@gmail.com)
 */

@Singleton
@Component(modules = {AppModule.class, BuildTypeModule.class})
public interface AppComponent {

    /**
     * @return {@link com.azizbekian.spyur.SpyurApplication SpyurApplication} context.
     */
     Context getApplicationContext();

    /**
     * @return Singleton {@link ApiInteractor} instance.
     */
     ApiInteractor getApiInteractor();

    /**
     * @return Singleton {@link RequestManager} instance.
     */
     RequestManager getGlide();

}

File: app/src/main/kotlin/com/incentive/yellowpages/injection/component/AppComponent.kt
package com.incentive.yellowpages.injection.component

import android.content.Context
import com.incentive.yellowpages.injection.ApplicationContext
import com.incentive.yellowpages.injection.module.AppModule
import com.incentive.yellowpages.injection.module.BuildTypeModule
import com.incentive.yellowpages.misc.ImageLoader
import dagger.Component
import javax.inject.Singleton

@Singleton
@Component(modules = arrayOf(AppModule::class, BuildTypeModule::class))
interface AppComponent {

    @ApplicationContext fun context(): Context

    fun imageLoader(): ImageLoader

    fun configPersistentComponent(): ConfigPersistentComponent

}
